elliott/elliottlib/cli/attach_cve_flaws_cli.py

```diff
@@ -1,23 +1,384 @@
+            )
+
+        # Get the shipment configs from the merge request URL
+        mr_url = shipment.get("url")
+        if not mr_url:
+            raise click.UsageError("Shipment block does not contain a 'url' field for the merge request")
+        self.logger.info(f"Fetching shipment configs from merge request: {mr_url}")
+
+        # Fetch the shipment configs from the merge request
+        release_notes = self.get_release_notes_from_mr(mr_url)
+
+        # Fetch the bug IDs from the release notes
+        bug_ids = [issue.id for issue in release_notes.issues.fixed] if release_notes.issues else []
+
+        if not bug_ids:
+            self.logger.info("No fixed issues found in the release notes, exiting.")
+            sys.exit(0)
+
+        # Get the bug trackers
+        tracker_bugs = self.get_attached_trackers(bug_ids, self.runtime.get_bug_tracker('jira'))
+        tracker_flaws, flaw_bugs = get_flaws(self.runtime.get_bug_tracker('bugzilla'), tracker_bugs)
+
+        if flaw_bugs:
+            self.logger.info(f"Found {len(flaw_bugs)} flaw bugs, updating release notes.")
+
+            # Turn the advisory type into an RHSA
+            self.update_konflux_release_notes_to_rhsa(release_notes, flaw_bugs, tracker_bugs, tracker_flaws)
+
+        return release_notes
+
+    def get_release_notes_from_mr(self, mr_url: str) -> ReleaseNotes:
+        """Fetch release notes from a merge request URL."""
+
+        kinds = (
+            (self.default_advisory_type,)
+            if self.default_advisory_type
+            else ("rpm", "image", "extras", "microshift", "metadata")
+        )
+        shipment_configs: Dict[str, ShipmentConfig] = get_shipment_configs_from_mr(mr_url, kinds)
+        shipment_config = shipment_configs.get(self.default_advisory_type)
+
+        if not shipment_config:
+            raise ValueError(f"No shipment config found for kind: {self.default_advisory_type}")
+
+        release_notes = shipment_config.shipment.data.releaseNotes
+        if not release_notes:
+            raise ValueError(f"No release notes found in shipment config for {self.default_advisory_type}")
+
+        return release_notes
+
+    def get_attached_trackers(self, bugs_ids: List[str], bug_tracker: BugTracker) -> List[Bug]:
+        """
+        Get attached tracker bugs from a list of bug IDs.
+        """
+
+        if not bugs_ids:
+            return []
+
+        attached_tracker_bugs: List[Bug] = bug_tracker.get_tracker_bugs(bugs_ids)
+        if not attached_tracker_bugs:
+            self.logger.info(f'Found 0 {bug_tracker.type} tracker bugs attached')
+            return []
+
+        self.logger.info(
+            f'Found {len(attached_tracker_bugs)} {bug_tracker.type} tracker bugs attached: '
+            f'{sorted([b.id for b in attached_tracker_bugs])}'
+        )
+        return attached_tracker_bugs
+
+    def update_konflux_release_notes_to_rhsa(
```

Comment:
I'm thinking it would be good to standardize names for brew and konflux, since right now it's quite hard to read the code and understand which methods belong to which build-system branch:
so would recommend, 
update_advisory -> update_advisory_brew() and 
update_konflux_release_notes_to_rhsa -> update_advisory_konflux

Reply:
Sounds good, done

Reply:
thank you!